From 968bf7d65812bba490c4222b3d05fa94cf856472 Mon Sep 17 00:00:00 2001
From: Stephanos Ioannidis <root@stephanos.io>
Date: Fri, 20 Dec 2019 11:18:46 +0900
Subject: [PATCH 6/6] Add newlib retargetable locking configuration

This commit adds support for the newlib configuration option
'--enable-newlib-retargetable-locking'.

Signed-off-by: Stephanos Ioannidis <root@stephanos.io>
---
 config/libc/newlib.in        | 14 ++++++++++++++
 scripts/build/libc/newlib.sh |  1 +
 2 files changed, 15 insertions(+)

diff --git a/config/libc/newlib.in b/config/libc/newlib.in
index bd193cd8..d84db70b 100644
--- a/config/libc/newlib.in
+++ b/config/libc/newlib.in
@@ -152,6 +152,13 @@ config LIBC_NEWLIB_MULTITHREAD
     help
         Enable support for multiple threads.
 
+config LIBC_NEWLIB_RETARGETABLE_LOCKING
+    bool
+    prompt "Enable retargetable locking"
+    help
+        Enable retargetable locking to allow the operating system to override
+        the dummy lock functions defined within the newlib.
+
 config LIBC_NEWLIB_EXTRA_SECTIONS
     bool
     prompt "Place each function & data element in their own section"
@@ -355,6 +362,13 @@ config LIBC_NANO_NEWLIB_MULTITHREAD
     help
         Enable support for multiple threads.
 
+config LIBC_NANO_NEWLIB_RETARGETABLE_LOCKING
+    bool
+    prompt "Enable retargetable locking"
+    help
+        Enable retargetable locking to allow the operating system to override
+        the dummy lock functions defined within the newlib.
+
 config LIBC_NANO_NEWLIB_EXTRA_SECTIONS
     bool
     prompt "Place each function & data element in their own section"
diff --git a/scripts/build/libc/newlib.sh b/scripts/build/libc/newlib.sh
index ca53c7f8..299698ac 100644
--- a/scripts/build/libc/newlib.sh
+++ b/scripts/build/libc/newlib.sh
@@ -220,6 +220,7 @@ GLOBAL_ATEXIT:newlib-global-atexit
 LITE_EXIT:lite-exit
 REENT_SMALL:newlib-reent-small
 MULTITHREAD:newlib-multithread
+RETARGETABLE_LOCKING:newlib-retargetable-locking
 WIDE_ORIENT:newlib-wide-orient
 UNBUF_STREAM_OPT:newlib-unbuf-stream-opt
 ENABLE_TARGET_OPTSPACE:target-optspace
-- 
2.17.1

